src/metpy/calc/thermo.py

```diff
@@ -3436,3 +3436,134 @@ def showalter_index(pressure, temperature, dewpoint):
 
     # Calculate the Showalter index
     return tp500 - prof[-1]
+
+
+@exporter.export
+@preprocess_and_wrap()
+@check_units('[pressure]', '[temperature]', '[temperature]')
+def total_totals_index(pressure, temperature, dewpoint):
+    """Calculate Total Totals Index from the pressure temperature and dewpoint.
+
+    Total Totals Index formula derived from [Miller1972]_:
+    TT = (T850 + Td850) - (2 * T500)
+
+    where:
+
+    T850 is the temperature at 850 hPa
+    T500 is the temperature at 500 hPa
+    Td850 is the dewpoint at 850 hPa
+
+    Calculation of the Total Totals Index is defined as the temperature at 850 hPa plus
+    the dewpoint at 850 hPa, minus twice the temperature at 500 hPa. This index consists of
+    two components, the Vertical Totals (VT) and the Cross Totals (CT).
+
+    Parameters
+    ----------
+    pressure : `pint.Quantity`
+        Pressure level(s), in order from highest to lowest pressure
+
+    temperature : `pint.Quantity`
+        Temperature corresponding to pressure
+
+    dewpoint : `pint.Quantity`
+        Dewpoint temperature corresponding to pressure
+
+    Returns
+    -------
+    `pint.Quantity`
+        Total Totals Index
+
+    """
+    # Find temperature and dewpoint at 850 and 500 hPa.
+    (t850, t500), (td850, _) = interpolate_1d(units.Quantity([850, 500], 'hPa'),
+                                              pressure, temperature, dewpoint)
+
+    # Calculate total totals index.
+    tt_index = ((t850.m + td850.m) * units.degC) - (2 * t500)
```

Comment:
This is incorrect unit-wise because it assumes `t850` and `td850` are in Celsius. Also, we don't want to multiply by units in the library, that breaks with masked arrays. Instead, try reformulating the calculation to properly give the sum of two temperature differences:
```suggestion
    tt_index = (t850 - t500) + (td850 - t500)
```
I'll also note that this implementation is exactly `tt_index = vertical_totals + cross_totals`.